tests/handover.cpp

```diff
@@ -0,0 +1,370 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "catch.hpp"
+
+#include "realm/commit_log.hpp"
+#include "util/test_file.hpp"
+
+#include "list.hpp"
+#include "object_accessor.hpp"
+#include "object_schema.hpp"
+#include "property.hpp"
+#include "schema.hpp"
+#include "handover.hpp"
+
+#include <thread>
+#include <future>
+
+using namespace realm;
+
+static TableRef get_table(Realm& realm, const ObjectSchema &object_schema) {
+    StringData name (("class_" + object_schema.name).c_str());
+    return realm.read_group().get_table(name);
+}
+
+static Object create_object(SharedRealm realm, const ObjectSchema &object_schema) {
+    TableRef table = get_table(*realm, object_schema);
+    return Object(std::move(realm), object_schema, Row((*table)[table->add_empty_row()]));
+}
+
+static List get_list(const Object& object, size_t column_ndx) {
+    return List(object.realm(), object.row().get_table()->get_linklist(column_ndx, object.row().get_index()));
+}
+
+static Property nullable(Property p) {
+    p.is_nullable = true;
+    return p;
+}
+
+TEST_CASE("handover") {
+    InMemoryTestFile config;
+    config.cache = false;
+    config.automatic_change_notifications = false;
+
+    SharedRealm r = Realm::get_shared_realm(config);
+
+    static const ObjectSchema string_object ({"string_object", {
+        nullable({"value", PropertyType::String}),
+    }});
+    static const ObjectSchema int_object ({"int_object", {
+        {"value", PropertyType::Int},
+    }});
+    static const ObjectSchema int_array_object ({"int_array_object", {
+        {"value", PropertyType::Array, "int_object"}
+    }});
+    r->update_schema({string_object, int_object, int_array_object});
+
+    SECTION("disallowed during write transactions") {
+        SECTION("export") {
+            r->begin_transaction();
+            REQUIRE_THROWS(r->package_for_handover({}));
+        }
+        SECTION("import") {
+            auto h = r->package_for_handover({});
+            r->begin_transaction();
+            REQUIRE_THROWS(r->accept_handover(std::move(h)));
+        }
+    }
+
+    SECTION("cleanup properly unpins version") {
+        auto durability = config.in_memory ? SharedGroup::durability_MemOnly :
+        SharedGroup::durability_Full;
```

Comment:
But since config will only ever be in-memory you should just pass that directly to the constructor.